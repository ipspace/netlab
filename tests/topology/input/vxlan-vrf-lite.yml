# Regression test for #643
#
# Also tests merging of modules introduced in 5f17c9b91fc3f4acdfd72d86b8b5097453f98936
#
defaults.device: eos

groups:
  hosts:
    members: [ rh1, rh2, rh3, bh1, bh2 ]
    module: []
    device: linux
  pe:
    module: [ vxlan,ospf ]
    members: [ s1,s2,s3 ]
  core:
    members: [ c ]
    module: [ ospf ]
  red_team:
    module: [ vlan,vrf ]
    members: [ s1, s2, s3 ]
    vlans:
      red_transport:
  blue_team:
    module: [ vlan,vrf ]
    members: [ s1, s2 ]
    vlans:
      blue_transport:

vrfs:
  red:
  blue:

vlans:
  red_transport:
    vrf: red
  blue_transport:
    vrf: blue

vxlan.vlans: [ red_transport, blue_transport ]

nodes: [ rh1, rh2, rh3, bh1, bh2, s1, s2, s3, c ]

links:
- s1-c
- s2-c
- s3-c

# Red VRF
- rh1:
  s1:
  vrf: red
- rh2:
  s2:
  vrf: red
- rh3:
  s3:
  vrf: red

# Blue VRF
- bh1:
  s1:
  vrf: blue
- bh2:
  s2:
  vrf: blue
