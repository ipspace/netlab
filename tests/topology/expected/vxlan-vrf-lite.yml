groups:
  blue_team:
    members:
    - s1
    - s2
    module:
    - vlan
    - vrf
    node_data:
      vlans:
        blue_transport: {}
  core:
    members:
    - c
    module:
    - ospf
  hosts:
    device: linux
    members:
    - rh1
    - rh2
    - rh3
    - bh1
    - bh2
    module: []
  pe:
    members:
    - s1
    - s2
    - s3
    module:
    - vxlan
    - ospf
  red_team:
    members:
    - s1
    - s2
    - s3
    module:
    - vlan
    - vrf
    node_data:
      vlans:
        red_transport: {}
input:
- topology/input/vxlan-vrf-lite.yml
- package:topology-defaults.yml
links:
- interfaces:
  - ifindex: 1
    ifname: Ethernet1
    ipv4: 10.1.0.2/30
    node: s1
  - ifindex: 1
    ifname: Ethernet1
    ipv4: 10.1.0.1/30
    node: c
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 10.1.0.0/30
  type: p2p
- interfaces:
  - ifindex: 1
    ifname: Ethernet1
    ipv4: 10.1.0.6/30
    node: s2
  - ifindex: 2
    ifname: Ethernet2
    ipv4: 10.1.0.5/30
    node: c
  linkindex: 2
  node_count: 2
  prefix:
    ipv4: 10.1.0.4/30
  type: p2p
- interfaces:
  - ifindex: 1
    ifname: Ethernet1
    ipv4: 10.1.0.10/30
    node: s3
  - ifindex: 3
    ifname: Ethernet3
    ipv4: 10.1.0.9/30
    node: c
  linkindex: 3
  node_count: 2
  prefix:
    ipv4: 10.1.0.8/30
  type: p2p
- bridge: input_4
  gateway:
    ipv4: 172.16.2.6/24
  interfaces:
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.2.1/24
    node: rh1
  - ifindex: 2
    ifname: Ethernet2
    ipv4: 172.16.2.6/24
    node: s1
  linkindex: 4
  node_count: 2
  prefix:
    ipv4: 172.16.2.0/24
  role: stub
  type: lan
  vrf: red
- bridge: input_5
  gateway:
    ipv4: 172.16.3.7/24
  interfaces:
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.3.2/24
    node: rh2
  - ifindex: 2
    ifname: Ethernet2
    ipv4: 172.16.3.7/24
    node: s2
  linkindex: 5
  node_count: 2
  prefix:
    ipv4: 172.16.3.0/24
  role: stub
  type: lan
  vrf: red
- bridge: input_6
  gateway:
    ipv4: 172.16.4.8/24
  interfaces:
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.4.3/24
    node: rh3
  - ifindex: 2
    ifname: Ethernet2
    ipv4: 172.16.4.8/24
    node: s3
  linkindex: 6
  node_count: 2
  prefix:
    ipv4: 172.16.4.0/24
  role: stub
  type: lan
  vrf: red
- bridge: input_7
  gateway:
    ipv4: 172.16.5.6/24
  interfaces:
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.5.4/24
    node: bh1
  - ifindex: 3
    ifname: Ethernet3
    ipv4: 172.16.5.6/24
    node: s1
  linkindex: 7
  node_count: 2
  prefix:
    ipv4: 172.16.5.0/24
  role: stub
  type: lan
  vrf: blue
- bridge: input_8
  gateway:
    ipv4: 172.16.6.7/24
  interfaces:
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.6.5/24
    node: bh2
  - ifindex: 3
    ifname: Ethernet3
    ipv4: 172.16.6.7/24
    node: s2
  linkindex: 8
  node_count: 2
  prefix:
    ipv4: 172.16.6.0/24
  role: stub
  type: lan
  vrf: blue
module:
- vlan
- ospf
- vrf
- vxlan
name: input
nodes:
  bh1:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 4
    interfaces:
    - bridge: input_7
      gateway:
        ipv4: 172.16.5.6/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.5.4/24
      linkindex: 7
      name: bh1 -> s1
      neighbors:
      - ifname: Ethernet3
        ipv4: 172.16.5.6/24
        node: s1
        vrf: blue
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.104
      mac: 08:4f:a9:00:00:04
    module: []
    name: bh1
    role: host
  bh2:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 5
    interfaces:
    - bridge: input_8
      gateway:
        ipv4: 172.16.6.7/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.6.5/24
      linkindex: 8
      name: bh2 -> s2
      neighbors:
      - ifname: Ethernet3
        ipv4: 172.16.6.7/24
        node: s2
        vrf: blue
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.105
      mac: 08:4f:a9:00:00:05
    module: []
    name: bh2
    role: host
  c:
    af:
      ipv4: true
    box: arista/veos
    device: eos
    id: 9
    interfaces:
    - ifindex: 1
      ifname: Ethernet1
      ipv4: 10.1.0.1/30
      linkindex: 1
      name: c -> s1
      neighbors:
      - ifname: Ethernet1
        ipv4: 10.1.0.2/30
        node: s1
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    - ifindex: 2
      ifname: Ethernet2
      ipv4: 10.1.0.5/30
      linkindex: 2
      name: c -> s2
      neighbors:
      - ifname: Ethernet1
        ipv4: 10.1.0.6/30
        node: s2
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    - ifindex: 3
      ifname: Ethernet3
      ipv4: 10.1.0.9/30
      linkindex: 3
      name: c -> s3
      neighbors:
      - ifname: Ethernet1
        ipv4: 10.1.0.10/30
        node: s3
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.9/32
      neighbors: []
      ospf:
        area: 0.0.0.0
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: Management1
      ipv4: 192.168.121.109
      mac: 08:4f:a9:00:00:09
    module:
    - ospf
    name: c
    ospf:
      af:
        ipv4: true
      area: 0.0.0.0
      router_id: 10.0.0.9
  rh1:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 1
    interfaces:
    - bridge: input_4
      gateway:
        ipv4: 172.16.2.6/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.2.1/24
      linkindex: 4
      name: rh1 -> s1
      neighbors:
      - ifname: Ethernet2
        ipv4: 172.16.2.6/24
        node: s1
        vrf: red
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    module: []
    name: rh1
    role: host
  rh2:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 2
    interfaces:
    - bridge: input_5
      gateway:
        ipv4: 172.16.3.7/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.3.2/24
      linkindex: 5
      name: rh2 -> s2
      neighbors:
      - ifname: Ethernet2
        ipv4: 172.16.3.7/24
        node: s2
        vrf: red
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    module: []
    name: rh2
    role: host
  rh3:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 3
    interfaces:
    - bridge: input_6
      gateway:
        ipv4: 172.16.4.8/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.4.3/24
      linkindex: 6
      name: rh3 -> s3
      neighbors:
      - ifname: Ethernet2
        ipv4: 172.16.4.8/24
        node: s3
        vrf: red
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.103
      mac: 08:4f:a9:00:00:03
    module: []
    name: rh3
    role: host
  s1:
    af:
      ipv4: true
      vpnv4: true
    box: arista/veos
    device: eos
    id: 6
    interfaces:
    - ifindex: 1
      ifname: Ethernet1
      ipv4: 10.1.0.2/30
      linkindex: 1
      name: s1 -> c
      neighbors:
      - ifname: Ethernet1
        ipv4: 10.1.0.1/30
        node: c
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    - bridge: input_4
      ifindex: 2
      ifname: Ethernet2
      ipv4: 172.16.2.6/24
      linkindex: 4
      name: s1 -> rh1
      neighbors:
      - ifname: eth1
        ipv4: 172.16.2.1/24
        node: rh1
      role: stub
      type: lan
      vrf: red
    - bridge: input_7
      ifindex: 3
      ifname: Ethernet3
      ipv4: 172.16.5.6/24
      linkindex: 7
      name: s1 -> bh1
      neighbors:
      - ifname: eth1
        ipv4: 172.16.5.4/24
        node: bh1
      role: stub
      type: lan
      vrf: blue
    - bridge_group: 1
      ifindex: 6
      ifname: Vlan1000
      ipv4: 172.16.0.6/24
      name: VLAN red_transport (1000) -> [s2,s3]
      neighbors:
      - ifname: Vlan1000
        ipv4: 172.16.0.7/24
        node: s2
      - ifname: Vlan1000
        ipv4: 172.16.0.8/24
        node: s3
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
      vrf: red
    - bridge_group: 2
      ifindex: 7
      ifname: Vlan1001
      ipv4: 172.16.1.6/24
      name: VLAN blue_transport (1001) -> [s2]
      neighbors:
      - ifname: Vlan1001
        ipv4: 172.16.1.7/24
        node: s2
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
      vrf: blue
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.6/32
      neighbors: []
      ospf:
        area: 0.0.0.0
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: Management1
      ipv4: 192.168.121.106
      mac: 08:4f:a9:00:00:06
    module:
    - vlan
    - ospf
    - vrf
    - vxlan
    name: s1
    ospf:
      af:
        ipv4: true
      area: 0.0.0.0
      router_id: 10.0.0.6
    vlan:
      max_bridge_group: 2
    vlans:
      blue_transport:
        bridge_group: 2
        id: 1001
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.1.0/24
        vni: 101001
        vrf: blue
        vtep_list:
        - 10.0.0.7
      red_transport:
        bridge_group: 1
        id: 1000
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
        vni: 101000
        vrf: red
        vtep_list:
        - 10.0.0.7
        - 10.0.0.8
    vrf:
      as: 65000
    vrfs:
      blue:
        af:
          ipv4: true
        export:
        - '65000:2'
        id: 2
        import:
        - '65000:2'
        ospf:
          active: true
          area: 0.0.0.0
          import:
            connected: {}
          interfaces:
          - bridge: input_7
            ifindex: 3
            ifname: Ethernet3
            ipv4: 172.16.5.6/24
            linkindex: 7
            name: s1 -> bh1
            neighbors:
            - ifname: eth1
              ipv4: 172.16.5.4/24
              node: bh1
            ospf:
              area: 0.0.0.0
              network_type: point-to-point
              passive: true
            role: stub
            type: lan
            vrf: blue
          - bridge_group: 2
            ifindex: 7
            ifname: Vlan1001
            ipv4: 172.16.1.6/24
            name: VLAN blue_transport (1001) -> [s2]
            neighbors:
            - ifname: Vlan1001
              ipv4: 172.16.1.7/24
              node: s2
            ospf:
              area: 0.0.0.0
              network_type: point-to-point
              passive: false
            type: svi
            virtual_interface: true
            vlan:
              mode: irb
            vrf: blue
          router_id: 10.0.0.6
        rd: '65000:2'
        vrfidx: 101
      red:
        af:
          ipv4: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        ospf:
          active: true
          area: 0.0.0.0
          import:
            connected: {}
          interfaces:
          - bridge: input_4
            ifindex: 2
            ifname: Ethernet2
            ipv4: 172.16.2.6/24
            linkindex: 4
            name: s1 -> rh1
            neighbors:
            - ifname: eth1
              ipv4: 172.16.2.1/24
              node: rh1
            ospf:
              area: 0.0.0.0
              network_type: point-to-point
              passive: true
            role: stub
            type: lan
            vrf: red
          - bridge_group: 1
            ifindex: 6
            ifname: Vlan1000
            ipv4: 172.16.0.6/24
            name: VLAN red_transport (1000) -> [s2,s3]
            neighbors:
            - ifname: Vlan1000
              ipv4: 172.16.0.7/24
              node: s2
            - ifname: Vlan1000
              ipv4: 172.16.0.8/24
              node: s3
            ospf:
              area: 0.0.0.0
              passive: false
            type: svi
            virtual_interface: true
            vlan:
              mode: irb
            vrf: red
          router_id: 10.0.0.6
        rd: '65000:1'
        vrfidx: 100
    vxlan:
      domain: global
      flooding: static
      vlans:
      - red_transport
      - blue_transport
      vtep: 10.0.0.6
      vtep_interface: Loopback0
      vtep_list:
      - 10.0.0.7
      - 10.0.0.8
  s2:
    af:
      ipv4: true
      vpnv4: true
    box: arista/veos
    device: eos
    id: 7
    interfaces:
    - ifindex: 1
      ifname: Ethernet1
      ipv4: 10.1.0.6/30
      linkindex: 2
      name: s2 -> c
      neighbors:
      - ifname: Ethernet2
        ipv4: 10.1.0.5/30
        node: c
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    - bridge: input_5
      ifindex: 2
      ifname: Ethernet2
      ipv4: 172.16.3.7/24
      linkindex: 5
      name: s2 -> rh2
      neighbors:
      - ifname: eth1
        ipv4: 172.16.3.2/24
        node: rh2
      role: stub
      type: lan
      vrf: red
    - bridge: input_8
      ifindex: 3
      ifname: Ethernet3
      ipv4: 172.16.6.7/24
      linkindex: 8
      name: s2 -> bh2
      neighbors:
      - ifname: eth1
        ipv4: 172.16.6.5/24
        node: bh2
      role: stub
      type: lan
      vrf: blue
    - bridge_group: 1
      ifindex: 6
      ifname: Vlan1000
      ipv4: 172.16.0.7/24
      name: VLAN red_transport (1000) -> [s1,s3]
      neighbors:
      - ifname: Vlan1000
        ipv4: 172.16.0.6/24
        node: s1
      - ifname: Vlan1000
        ipv4: 172.16.0.8/24
        node: s3
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
      vrf: red
    - bridge_group: 2
      ifindex: 7
      ifname: Vlan1001
      ipv4: 172.16.1.7/24
      name: VLAN blue_transport (1001) -> [s1]
      neighbors:
      - ifname: Vlan1001
        ipv4: 172.16.1.6/24
        node: s1
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
      vrf: blue
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.7/32
      neighbors: []
      ospf:
        area: 0.0.0.0
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: Management1
      ipv4: 192.168.121.107
      mac: 08:4f:a9:00:00:07
    module:
    - vlan
    - ospf
    - vrf
    - vxlan
    name: s2
    ospf:
      af:
        ipv4: true
      area: 0.0.0.0
      router_id: 10.0.0.7
    vlan:
      max_bridge_group: 2
    vlans:
      blue_transport:
        bridge_group: 2
        id: 1001
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.1.0/24
        vni: 101001
        vrf: blue
        vtep_list:
        - 10.0.0.6
      red_transport:
        bridge_group: 1
        id: 1000
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
        vni: 101000
        vrf: red
        vtep_list:
        - 10.0.0.6
        - 10.0.0.8
    vrf:
      as: 65000
    vrfs:
      blue:
        af:
          ipv4: true
        export:
        - '65000:2'
        id: 2
        import:
        - '65000:2'
        ospf:
          active: true
          area: 0.0.0.0
          import:
            connected: {}
          interfaces:
          - bridge: input_8
            ifindex: 3
            ifname: Ethernet3
            ipv4: 172.16.6.7/24
            linkindex: 8
            name: s2 -> bh2
            neighbors:
            - ifname: eth1
              ipv4: 172.16.6.5/24
              node: bh2
            ospf:
              area: 0.0.0.0
              network_type: point-to-point
              passive: true
            role: stub
            type: lan
            vrf: blue
          - bridge_group: 2
            ifindex: 7
            ifname: Vlan1001
            ipv4: 172.16.1.7/24
            name: VLAN blue_transport (1001) -> [s1]
            neighbors:
            - ifname: Vlan1001
              ipv4: 172.16.1.6/24
              node: s1
            ospf:
              area: 0.0.0.0
              network_type: point-to-point
              passive: false
            type: svi
            virtual_interface: true
            vlan:
              mode: irb
            vrf: blue
          router_id: 10.0.0.7
        rd: '65000:2'
        vrfidx: 101
      red:
        af:
          ipv4: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        ospf:
          active: true
          area: 0.0.0.0
          import:
            connected: {}
          interfaces:
          - bridge: input_5
            ifindex: 2
            ifname: Ethernet2
            ipv4: 172.16.3.7/24
            linkindex: 5
            name: s2 -> rh2
            neighbors:
            - ifname: eth1
              ipv4: 172.16.3.2/24
              node: rh2
            ospf:
              area: 0.0.0.0
              network_type: point-to-point
              passive: true
            role: stub
            type: lan
            vrf: red
          - bridge_group: 1
            ifindex: 6
            ifname: Vlan1000
            ipv4: 172.16.0.7/24
            name: VLAN red_transport (1000) -> [s1,s3]
            neighbors:
            - ifname: Vlan1000
              ipv4: 172.16.0.6/24
              node: s1
            - ifname: Vlan1000
              ipv4: 172.16.0.8/24
              node: s3
            ospf:
              area: 0.0.0.0
              passive: false
            type: svi
            virtual_interface: true
            vlan:
              mode: irb
            vrf: red
          router_id: 10.0.0.7
        rd: '65000:1'
        vrfidx: 100
    vxlan:
      domain: global
      flooding: static
      vlans:
      - red_transport
      - blue_transport
      vtep: 10.0.0.7
      vtep_interface: Loopback0
      vtep_list:
      - 10.0.0.6
      - 10.0.0.8
  s3:
    af:
      ipv4: true
      vpnv4: true
    box: arista/veos
    device: eos
    id: 8
    interfaces:
    - ifindex: 1
      ifname: Ethernet1
      ipv4: 10.1.0.10/30
      linkindex: 3
      name: s3 -> c
      neighbors:
      - ifname: Ethernet3
        ipv4: 10.1.0.9/30
        node: c
      ospf:
        area: 0.0.0.0
        network_type: point-to-point
        passive: false
      type: p2p
    - bridge: input_6
      ifindex: 2
      ifname: Ethernet2
      ipv4: 172.16.4.8/24
      linkindex: 6
      name: s3 -> rh3
      neighbors:
      - ifname: eth1
        ipv4: 172.16.4.3/24
        node: rh3
      role: stub
      type: lan
      vrf: red
    - bridge_group: 1
      ifindex: 4
      ifname: Vlan1000
      ipv4: 172.16.0.8/24
      name: VLAN red_transport (1000) -> [s1,s2]
      neighbors:
      - ifname: Vlan1000
        ipv4: 172.16.0.6/24
        node: s1
      - ifname: Vlan1000
        ipv4: 172.16.0.7/24
        node: s2
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
      vrf: red
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.8/32
      neighbors: []
      ospf:
        area: 0.0.0.0
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: Management1
      ipv4: 192.168.121.108
      mac: 08:4f:a9:00:00:08
    module:
    - vlan
    - ospf
    - vrf
    - vxlan
    name: s3
    ospf:
      af:
        ipv4: true
      area: 0.0.0.0
      router_id: 10.0.0.8
    vlan:
      max_bridge_group: 1
    vlans:
      red_transport:
        bridge_group: 1
        id: 1000
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
        vni: 101000
        vrf: red
        vtep_list:
        - 10.0.0.6
        - 10.0.0.7
    vrf:
      as: 65000
    vrfs:
      red:
        af:
          ipv4: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        ospf:
          active: true
          area: 0.0.0.0
          import:
            connected: {}
          interfaces:
          - bridge: input_6
            ifindex: 2
            ifname: Ethernet2
            ipv4: 172.16.4.8/24
            linkindex: 6
            name: s3 -> rh3
            neighbors:
            - ifname: eth1
              ipv4: 172.16.4.3/24
              node: rh3
            ospf:
              area: 0.0.0.0
              network_type: point-to-point
              passive: true
            role: stub
            type: lan
            vrf: red
          - bridge_group: 1
            ifindex: 4
            ifname: Vlan1000
            ipv4: 172.16.0.8/24
            name: VLAN red_transport (1000) -> [s1,s2]
            neighbors:
            - ifname: Vlan1000
              ipv4: 172.16.0.6/24
              node: s1
            - ifname: Vlan1000
              ipv4: 172.16.0.7/24
              node: s2
            ospf:
              area: 0.0.0.0
              passive: false
            type: svi
            virtual_interface: true
            vlan:
              mode: irb
            vrf: red
          router_id: 10.0.0.8
        rd: '65000:1'
        vrfidx: 100
    vxlan:
      domain: global
      flooding: static
      vlans:
      - red_transport
      vtep: 10.0.0.8
      vtep_interface: Loopback0
      vtep_list:
      - 10.0.0.6
      - 10.0.0.7
ospf:
  area: 0.0.0.0
provider: libvirt
vlans:
  blue_transport:
    host_count: 0
    id: 1001
    neighbors:
    - ifname: Vlan1001
      ipv4: 172.16.1.6/24
      node: s1
    - ifname: Vlan1001
      ipv4: 172.16.1.7/24
      node: s2
    prefix:
      allocation: id_based
      ipv4: 172.16.1.0/24
    vni: 101001
    vrf: blue
  red_transport:
    host_count: 0
    id: 1000
    neighbors:
    - ifname: Vlan1000
      ipv4: 172.16.0.6/24
      node: s1
    - ifname: Vlan1000
      ipv4: 172.16.0.7/24
      node: s2
    - ifname: Vlan1000
      ipv4: 172.16.0.8/24
      node: s3
    prefix:
      allocation: id_based
      ipv4: 172.16.0.0/24
    vni: 101000
    vrf: red
vrf:
  as: 65000
vrfs:
  blue:
    export:
    - '65000:2'
    id: 2
    import:
    - '65000:2'
    rd: '65000:2'
  red:
    export:
    - '65000:1'
    id: 1
    import:
    - '65000:1'
    rd: '65000:1'
vxlan:
  domain: global
  flooding: static
  use_v6_vtep: false
  vlans:
  - red_transport
  - blue_transport
